### PR TITLE
DSD-859: Refactoring RadioGroup

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-859](https://jira.nypl.org/browse/DSD-859)

## This PR does the following:

- Refactors the `RadioGroup` to use Chakra's `RadioGroup` component rather than the `useRadioGroup` hook. This is cleaner and it's easier to reason out the props.
- Adds internal state to keep track of the selected value.
- Updates the example in the `Form.stories.mdx` file.

The "uncontrolled" version of Chakra's `RadioGroup` is not working and I want to investigate this in the future. This means that the DS' `RadioGroup` won't work through "ref"s. This may not be a problem any time soon but might be in the future.

I tested this by using Chakra's `Radio` and `RadioGroup` components directly in examples and passing in a ref. Trying to get the value when submitting did not work. I need to try in other and new environments but this may be a Chakra bug.

## How has this been tested?

Storybook

## Accessibility concerns or updates

- N/A

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] View [the example in Storybook]()

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr920-uwohlddxzhango4kisekhl6wi1tirzka.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/623e313b97feaa22ee131dd9

**Review by bigfishdesign13 — COMMENTED** [review]
This looks good in Chrome, but I am seeing something odd in both `CheckboxGroup` and `RadioGroup` in Safari.  The `option` text is bouncing onto a separate line and I am not sure why.  The CSS looks good and Safari supports the `justify-items` attribute.

image

Can you check that as well?

**Comment by EdwinGuzman** [comment]
@bigfishdesign13 updated so the `legend` now has the CSS `::after` pseudo "content" style rule. This only  affects `CheckboxGroup`, `DatePicker`, and `RadioGroup`. This is to make it work in Safari. Adding notes here I found and things we discussed on slack:

- This codeSandbox shows that "div"s and "span"s render correctly in a "legend" element that has "display: flex" https://codesandbox.io/s/styled-components-forked-ryr6od?file=/index.js. This is not true in Safari as divs render as block elements and span don't render at all. This is only a problem with the "legend" element and not a div wrapper. In our case, we _want_ to use the "legend" element since it's part of the fieldset wrapper.
- The fieldset element does have some issues with styling: https://bugs.webkit.org/show_bug.cgi?id=220793
- Even though ":after" CSS pseudo rule works to display the "Optional"/"Required" text in the legend, it's not accessible and it's hard to test because of how jsdom works. 


Update: Most of this won't be true with the next label/legend update in the DS. Will put this PR as "blocked" until that work is done.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr920-uwohlddxzhango4kisekhl6wi1tirzka.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/623e313b97feaa22ee131dd9

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ Currently, this repo is in Prerelease. When it is released, this project will ad
 
 - Exports the `useCarouselStyles` and `useWindowSize` hooks and adds documentation for all hooks in Storybook.
 - Adds additional semantic design tokens from `fontWeights` and `fontSizes` to the `useNYPLTheme` hook.
+- Adds a warning if a required `id` prop is not passed to one of the "Form Elements": `Button`, `Checkbox`, `CheckboxGroup`, `DatePicker`, `Fieldset`, `Form`, `Label`, `ProgressIndicator`, `Radio`, `RadioGroup`, `SearchBar`, `Select`, `Slider`, `TextInput`, `Toggle`.
```
Comment (EdwinGuzman) [inline CHANGELOG.md:15]: Some comments were under the previous release and had to be moved up.

**Comment by EdwinGuzman** [comment]
No longer blocked and updated.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr920-uwohlddxzhango4kisekhl6wi1tirzka.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/623e313b97feaa22ee131dd9

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr920-uwohlddxzhango4kisekhl6wi1tirzka.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/623e313b97feaa22ee131dd9

**Review by bigfishdesign13 — APPROVED** [review]
Looks good.

**Pull request merged**